data-prepper-plugins/anomaly-detector-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/AnomalyDetectorMode.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.anomalydetector;
+
+import java.util.List;
+import java.util.Collection;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+
+public interface AnomalyDetectorMode {
+    public void initialize(List<String> keys);
+    public Collection<Record<Event>> handleEvents(Collection<Record<Event>> record);
```

Comment:
Let's add javadocs to this interface describing the functionality that should be implemented and used.